samtranslator/translator/translator.py

```diff
@@ -31,6 +31,39 @@ def __init__(self, managed_policy_map, sam_parser, plugins=None):
         self.plugins = plugins
         self.sam_parser = sam_parser
 
+    function_names = dict()
+
+    def _get_function_names(self, resource_dict, parameter_values):
```

Comment:
This function appends all the function names of an API as an API can have multiple functions. If atleast one function name changes, the api will be redeployed.